app-ios/tutanota/Sources/AppDelegate.swift

```diff
@@ -1,158 +1,123 @@
+		let userPreferencesProvider = UserPreferencesProviderImpl()
+		let notificationStorage = NotificationStorage(userPreferencesProvider: userPreferencesProvider)
+		let keychainManager = KeychainManager(keyGenerator: KeyGenerator())
+
+		let alarmModel = AlarmModel(dateProvider: SystemDateProvider())
+		self.alarmManager = AlarmManager(
+			alarmPersistor: AlarmPreferencePersistor(notificationsStorage: notificationStorage, keychainManager: keychainManager),
+			alarmCryptor: KeychainAlarmCryptor(keychainManager: keychainManager), alarmScheduler: SystemAlarmScheduler(), alarmCalculator: alarmModel)
+		self.notificationsHandler = NotificationsHandler(alarmManager: self.alarmManager, notificationStorage: notificationStorage)
+		self.window = UIWindow(frame: UIScreen.main.bounds)
+		let credentialsEncryption = IosNativeCredentialsFacade(keychainManager: keychainManager)
+
+		self.viewController = ViewController(
+			crypto: IosNativeCryptoFacade(), themeManager: ThemeManager(userProferencesProvider: userPreferencesProvider), keychainManager: keychainManager,
+			notificationStorage: notificationStorage, alarmManager: alarmManager, notificaionsHandler: notificationsHandler,
+			credentialsEncryption: credentialsEncryption, blobUtils: BlobUtil(),
+			contactsSynchronization: ContactsSynchronization(userDefault: UserDefaults.standard))
+		self.window!.rootViewController = viewController
+
+		UNUserNotificationCenter.current().delegate = self
+
+		window!.makeKeyAndVisible()
+	}
+
+	func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+		TUTSLog("Start Tutanota with launch options: \(String(describing: launchOptions))")
+		self.start()
+		return true
+	}
+
+	func applicationWillEnterForeground(_ application: UIApplication) { UIApplication.shared.applicationIconBadgeNumber = 0 }
+
+	func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
+		if let callback = self.pushTokenCallback {
+			let stringToken = deviceTokenAsString(deviceToken: deviceToken)
+			callback(.success(stringToken!))
+			self.pushTokenCallback = nil
+		}
+	}
+
+	func application(_ application: UIApplication, didFailToRegisterForRemoteNotificationsWithError error: Error) {
+		self.pushTokenCallback?(.failure(error))
+		self.pushTokenCallback = nil
+	}
+
+	/// handles tutanota deep links:
+	/// tutanota:// -> ?
+	/// tutashare:// -> share requests from the sharing extension
+	func application(_ application: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey: Any] = [:]) -> Bool {
+		switch url.scheme {
+		case TUTANOTA_SHARE_SCHEME: Task { try! await self.viewController.handleShare(url) }
+		case nil: TUTSLog("missing scheme!")
+		default: TUTSLog("unknown scheme? \(url.scheme!)")
+		}
+		return true
+	}
+
+	func application(
+		_ application: UIApplication, didReceiveRemoteNotification userInfo: [AnyHashable: Any],
```

Comment:
also this. I would prefer if it would break line before each argument if they don't fit onto one line. I think this is what `lineBreakBeforeEachArgument` does?

Reply:
Much better